pkg/reconciler/app/resources/route.go

```diff
@@ -38,12 +96,42 @@ func MakeRoutes(app *v1alpha1.App, space *v1alpha1.Space) ([]v1alpha1.Route, err
 
 		routes = append(routes, v1alpha1.Route{
 			ObjectMeta: metav1.ObjectMeta{
-				Name:      v1alpha1.GenerateRouteName(appRoute.Hostname, appRoute.Domain, appRoute.Path),
+				Name: v1alpha1.GenerateRouteName(
+					appRoute.Hostname,
+					appRoute.Domain,
+					appRoute.Path,
+					app.Name,
+				),
+				Namespace: space.Name,
+				Labels: resources.UnionMaps(
+					app.GetLabels(),
```

Comment:
It would be great to add in the app.ComponentLabels here with a type of "route" https://github.com/google/kf/blob/master/pkg/apis/kf/v1alpha1/app_types.go#L60

These can be used by the Application CRD to group together bits and pieces of infrastructure into a single UI like GKE's.

Reply:
I actually had that at one point, however in CF, a route can have multiple apps routed to it. We are wanting to move to that as well (#279).

Reply:
I think to keep the ownership model simple, the Route (cf's route mapping) object should belong to an app and we should allow multiple apps to have Routes that point to the same URI. The RouteClaim (cf's route) could be a pooling mechanism to do fast lookups like listing all the routes in the space or the apps associated with a route.

WDYT?

Reply:
So actually I'm going to say my last comment is incorrect. With this model, each Route DOES exist for an app and the Controller is aggregating them.